### ISSUE TITLE
InlinePanel will display redundant fields when using option 'panels'

### ISSUE THREAD

**Issue by kidlj** [torchbox/wagtail#922]
As in 'models.py' of "wagtaildemo" app, but with `panels` option in use for InlinePanel and set to `LinkFields.panels`(note: not to `CarouselItem.panels`).

```
HomePage.content_panels = [
    # ...
    InlinePanel(HomePage, 'Carousel_items', panels=LinkFields.panels, label="Carousel items")
```
In this case, it will display(see attachment):
![inline_panels](https://cloud.githubusercontent.com/assets/300616/5854917/a50bdbdc-a26d-11e4-86af-b726fe11d551.png)

**Comment by gasman** [comment]
Thanks for the report! This is a variant of the bug reported in #338 - I'm working on a fix which should solve this issue too.

**Comment by gasman** [comment]
Fixed in #957
